src/internal/packager/compose.go

```diff
@@ -42,50 +46,105 @@ func validateOrBail(component *types.ZarfComponent) {
 }
 
 // Sets Name, Default, Required and Description to the original components values
-func mergeComponentOverrides(target *types.ZarfComponent, src types.ZarfComponent) {
-	target.Name = src.Name
-	target.Default = src.Default
-	target.Required = src.Required
+func mergeComponentOverrides(target *types.ZarfComponent, override types.ZarfComponent) {
+	target.Name = override.Name
+	target.Default = override.Default
+	target.Required = override.Required
+	target.Group = override.Group
+
+	// Override description if it was provided.
+	if override.Description != "" {
+		target.Description = override.Description
+	}
+
+	// Override cosign key path if it was provided.
+	if override.CosignKeyPath != "" {
+		target.CosignKeyPath = override.CosignKeyPath
+	}
+
+	// Append slices where they exist.
+	target.Charts = append(target.Charts, override.Charts...)
+	target.DataInjections = append(target.DataInjections, override.DataInjections...)
+	target.Files = append(target.Files, override.Files...)
+	target.Images = append(target.Images, override.Images...)
+	target.Manifests = append(target.Manifests, override.Manifests...)
+	target.Repos = append(target.Repos, override.Repos...)
+
+	// Merge variables.
+	for key, variable := range override.Variables {
+		target.Variables[key] = variable
+	}
 
-	if src.Description != "" {
-		target.Description = src.Description
+	// Merge scripts.
+	target.Scripts.Before = append(target.Scripts.Before, override.Scripts.Before...)
```

Comment:
@jeff-mccoy I'm ~~pretty sure~~ _really hoping_ the k3s tests are almost always failing on this branch because we aren't merging the `retry: true` value to the new k3s component. Since it's not retrying it's failing really early because the OS isn't ready to start up the k3s service the first couple 'attempts'.

Reply:
hmmm looks like retry is still running after all?